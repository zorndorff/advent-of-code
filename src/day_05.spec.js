const day = require('./day_5');
const {} = require('./functions');

const input_day_05 = `0,9 -> 5,9
8,0 -> 0,8
9,4 -> 3,4
2,2 -> 2,1
7,0 -> 7,4
6,4 -> 2,0
0,9 -> 2,9
3,4 -> 1,4
0,0 -> 8,8
5,5 -> 8,2`;



describe('day 05', () => {
  it('parseInput: parses input into list of coordinate segments', () => {
    const result = day.parseInput(input_day_05.split('\n'));
    expect(result).toMatchSnapshot();
  });
  it('process: returns the correct number of overlapping points', () => {
    const result = day.process(input_day_05.split('\n'));
    expect(result.overlapPoints).toEqual(5);
  })

});
