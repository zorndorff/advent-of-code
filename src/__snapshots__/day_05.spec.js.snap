// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`day 05 parseInput: parses input into list of coordinate segments 1`] = `
Object {
  "grid": Array [
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
    Array [
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
      0,
    ],
  ],
  "lineSegments": Array [
    Array [
      Array [
        0,
        9,
      ],
      Array [
        5,
        9,
      ],
    ],
    Array [
      Array [
        8,
        0,
      ],
      Array [
        0,
        8,
      ],
    ],
    Array [
      Array [
        9,
        4,
      ],
      Array [
        3,
        4,
      ],
    ],
    Array [
      Array [
        2,
        2,
      ],
      Array [
        2,
        1,
      ],
    ],
    Array [
      Array [
        7,
        0,
      ],
      Array [
        7,
        4,
      ],
    ],
    Array [
      Array [
        6,
        4,
      ],
      Array [
        2,
        0,
      ],
    ],
    Array [
      Array [
        0,
        9,
      ],
      Array [
        2,
        9,
      ],
    ],
    Array [
      Array [
        3,
        4,
      ],
      Array [
        1,
        4,
      ],
    ],
    Array [
      Array [
        0,
        0,
      ],
      Array [
        8,
        8,
      ],
    ],
    Array [
      Array [
        5,
        5,
      ],
      Array [
        8,
        2,
      ],
    ],
  ],
}
`;
